working/augmentation-libraries/feature-specification.md

```diff
@@ -869,49 +879,189 @@ It is a compile-time error if:
+*   For each entry in the initializer list of *d*, in sequence order:
 
-    *   Otherwise the result of applying the augmenting constructor has an
-        initializer list containing first the assertions and field initializers
-        of the augmented constructor, if any, then the assertions and field
-        initializers of the augmenting constructor, and finally any
-        super-initializer of either the augmeted or augmenting constructor.
+    *   If assert entry, evaluate the first operand in the initializer list
+        scope. If evaluate to `false`, evaluate the second operand to a message
+        value *m*, if there is a second operand, otherwise let *m* be the
+        `null` value. Throw an `AssertionError` with *m* as message.
+
+    *   If variable intializer entry, evaluate expression in initializer list
+        scope to value *v*. Initialize the variable of *o* corresponding to
+        variable with the initializer entry name in *C* to the value *v*.
+
+*   If *d* has a *superConstructorInvocation*.
+
+    * It is an error if *s* is already initialized.
+
+    * Initialize *s* to the *superConstructorInvocation* of *d*.
+
+*   If *d* has an *augmentedDefinition* *d’*
+
+    *   Invoke *d’* on *C\<T1,…,Tn>* with arguments *L*, super-arguments *S*,
+        and super constructor invocation *s* to initialize *o*.
+
+    *   *This recurses on the augmented definition*.
+
+*   Otherwise:
+
+    *   Let *U* be be the instantiated *superclass definition* of
+        *C*\<*T1*,…,*Tn*\>.
+
+    *   This may be a synthetic definition (for an anonymous mixin application)
+        computed from the declared superclass and declared mixins of *C*. _(Or
+        we can do the short-circuiting here since mixin applications can only)
+
+    *   If *s* is initialized:
+
+        *   Evaluate each argument of the argument list of the super-constructor
+            invocation in the initializer list scope, in source order, and set
+            the entry of *S* with the same position or name to the resulting
+            value.
+
+        *   Let *g* be the name of superclass constructor of *U* targeted by the
+            *superConstructorInvocation* (class-name of *U* plus `.id` if
+            referenced as  `super.id`).
+
+    *   Otherwise let *g* be the name of the unnamed constructor of *U*.
+
+    *   Invoke the constructor named *g* on *U* with arguments *S* to initialize
+        *o*.
+
+    *   *This recurses on the superclass constructor.*
+
+*   When this has completed, execute the *body* of *d*, if any, in a scope which
+    has the parameter scope of the invocation of *d* as parent scope.
+
+*   Then invocation of *d* completes normally.
+
+The consequence of this definition is an execution order for initialization of
+an instance of a class of:
+
+*   Field initializers of class, from all augmentations, in “base declaration
+    source order”.
+
+*   Augmenting declaration parameter lists and initializer lists, for
```

Comment:
The "evaluate parameter list" here is actually performing "binding actuals to formals", which is the step that handles initializing formals and super parameters.

Or in more detail... When you execute a (non-redirecting generative) contructor of a class to initialize the object, given an argument list, it does the following:
- For every instance variable of the class, in source order of their base declarations,
  execute any initializer expressions for the variable.
  - Start with the last variable declaration for that variable that has an initializing expression.
  - If that declaration is in an augmenting declaration, and it contains `augmented`,
    then each occurrence of `augmented` evaluates the initializer expression of the
    next prior declaration of that variable that has an initializing expression.
  - Then initialize the variable to the resulting value.
- Then start with the last (in source order) declaration of that constructor, and execute that
  to initialize the new object with the given argument list.
- Executing a constructor declaration in this way means:
    - Bind the actual argument list to the declared formal parameters of that constructor 
      declaration. This handles initializing formals and super parameters (which assign the 
      value to some position in a super constructor invocation argument list), and binds
      names to values in the parameter and initializer list scopes.
    - Execute the initializer list of that constructor declaration in the intializer list scope.
    - If the constructor declares a super-constructor invocation, remember that superclass
      constructor, and the accumulated super-constructor argument list.
    - If the declaration is an augmenting constructor declaration, then
      execute (recursively) the augmented declaration to initialize the same object
      with the same argument list.
     - Otherwise execute the remembered superclass constructor with the constructed
       argument list to initialize the same object. (If there is no remembered superclass
       constructor, execute the unnamed constructor of the superclass with the collected
       argument list).
    - When either of these complete, execute the body (if any) of this declaration in the 
      parameter scope of.
    - Then the execution of this constructor declaration completes.
- When the last constructor declaration execution has completed, the execution of the
  class constructor completes.